Caboodle/Magnetometer/Magnetometer.uwp.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Windows.Devices.Sensors;
+using WindowsMagnetometer = Windows.Devices.Sensors.Magnetometer;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Magnetometer
+    {
+        // Magic numbers from https://docs.microsoft.com/en-us/uwp/api/windows.devices.sensors.compass.reportinterval#Windows_Devices_Sensors_Compass_ReportInterval
+        internal const uint FastestInterval = 8;
+        internal const uint GameInterval = 22;
+        internal const uint NormalInterval = 33;
+
+        internal static WindowsMagnetometer DefaultMagnetometer =>
+            WindowsMagnetometer.GetDefault();
+
+        internal static bool IsSupported =>
+            DefaultMagnetometer != null;
+
+        internal static void PlatformStart(SensorSpeed sensorSpeed)
+        {
+            var sensor = DefaultMagnetometer;
+            var interval = NormalInterval;
+            switch (sensorSpeed)
+            {
+                case SensorSpeed.Fastest:
+                    interval = FastestInterval;
+                    break;
+                case SensorSpeed.Game:
+                    interval = GameInterval;
+                    break;
+            }
+
+            sensor.ReportInterval = sensor.MinimumReportInterval >= interval ? interval : sensor.MinimumReportInterval;
```

Comment:
Just checking if this is the right way: `min >= desired ? desired : min`

Reply:
Is min a _high_ number, or a low number?

Reply:
Ah, I see when I reversed it... maybe flip it around to `min < interval ? min : interval` as it just appears weird.

Reply:
No wait... that also looks weird.

Reply:
#resolved

Reply:
https://docs.microsoft.com/en-us/windows/uwp/devices-sensors/use-the-compass